query/src/main/java/org/infinispan/query/impl/massindex/IndexWorker.java

```diff
@@ -151,7 +153,15 @@ private Object extractValue(Object storageValue, DataConversion valueDataConvers
       @Override
       public void writeObject(ObjectOutput output, IndexWorker worker) throws IOException {
          output.writeObject(worker.cacheName);
-         output.writeObject(PojoIndexedTypeIdentifier.convertToLegacy(worker.indexedType));
+         if (worker.indexedTypes == null) {
+            output.writeInt(0);
+         } else {
+            Set<? extends Class<?>> types = worker.indexedTypes.stream().map(PojoIndexedTypeIdentifier::convertToLegacy).collect(Collectors.toSet());
```

Comment:
This ```types``` is used only to count them ? And then writeObject + convertToLegacy again? surely a mistake

Reply:
ops...